### PR TITLE
NAS refactor initialization

### PR DESCRIPTION
**Pull request by ultmaster**

### CODE REVIEW

File: src/sdk/pynni/nni/nas/pytorch/darts/trainer.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+import torch
+from torch import nn as nn
+
+from nni.nas.pytorch.trainer import Trainer
+from nni.nas.utils import AverageMeterGroup, auto_device
+from .mutator import DartsMutator
+
+
+class DartsTrainer(Trainer):
+    def __init__(self, model, loss, metrics,
```
Comment (xuehui1991) [inline src/sdk/pynni/nni/nas/pytorch/darts/trainer.py]: docstring?

File: src/sdk/pynni/nni/nas/pytorch/darts/trainer.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+            self.train_epoch(epoch)
+
+            # validation
+            print("Epoch {} Validating".format(epoch))
+            self.validate_epoch(epoch)
+
+    def _unrolled_backward(self, trn_X, trn_y, val_X, val_y, backup_model, lr):
+        """
+        Compute unrolled loss and backward its gradients
+        Parameters
```
Comment (xuehui1991) [inline src/sdk/pynni/nni/nas/pytorch/darts/trainer.py]: add blank line after line 114.

File: examples/nas/darts/main.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+                           metrics=lambda output, target: accuracy(output, target, topk=(1,)),
+                           model_optim=optim,
+                           lr_scheduler=lr_scheduler,
+                           num_epochs=50,
+                           dataset_train=dataset_train,
+                           dataset_valid=dataset_valid,
+                           batch_size=args.batch_size,
+                           log_frequency=args.log_frequency)
+    trainer.train()
+    trainer.finalize()
```
Comment (QuanluZhang) [inline examples/nas/darts/main.py]: please change `finalize` to `export`

File: src/sdk/pynni/nni/nas/pytorch/darts/mutator.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import torch
+from torch import nn as nn
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/nas/pytorch/darts/mutator.py]: nn as nn?

**Pull request merged**